infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -693,15 +695,25 @@ private void readAddressSpace(SegmentHandle segment) throws IOException {
     private LogData readRecord(SegmentHandle segment, long address) throws IOException {
         FileChannel fileChannel = segment.getReadChannel();
 
-        AddressMetaData metaData = segment.getKnownAddresses().get(address);
-        if (metaData == null) {
+        if (!segment.getKnownAddresses().containsKey(address)) {
             return null;
         }
 
         try {
-            ByteBuffer entryBuf = ByteBuffer.allocate(metaData.length);
-            fileChannel.read(entryBuf, metaData.offset);
+            long metaDataOffSet = segment.getKnownAddresses().get(address);
+            // TODO(Maithem): Pre-allocate a large buffer to avoid the second read
+            ByteBuffer metaData = ByteBuffer.allocate(METADATA_SIZE);
```

Comment:
Please change metaData here to metaDataBuffer. As the same variable name is used again for MetaData class object two lines below.

Reply:
ok